### PR TITLE
move away from shrinkwrapsharedserver

### PR DESCRIPTION
**Pull request by benjamin-confino**
This is part of https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.workitem.viewWorkItem&id=279982 

Unlike my earlier impression moving from the outdated framework wont fix the issue, but its done now and good to move away from depreciated code.

### CODE REVIEW

File: dev/com.ibm.ws.cdi.beansxml_fat/fat/src/com/ibm/ws/cdi12/fat/tests/BeanDiscoveryModeNoneTest.java [hunk 1]
```diff
@@ -26,28 +27,34 @@
...
-    public static ShrinkWrapSharedServer SHARED_SERVER = new ShrinkWrapSharedServer("cdi12BeanDiscoveryModeNoneServer");
+    private static LibertyServer server;
 
-    @BuildShrinkWrap
-    public static Archive buildShrinkWrap() {
-         
-         return ShrinkWrap.create(WebArchive.class, "beanDiscoveryModeNone.war")
+    @BeforeClass
+    public static void setUp() throws Exception {
+         server = LibertyServerFactory.getLibertyServer("cdi12BeanDiscoveryModeNoneServer");
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.beansxml_fat/fat/src/com/ibm/ws/cdi12/fat/tests/BeanDiscoveryModeNoneTest.java:50]: You can take this line out and add `@Server` to the `LibertyServer` field instead.

File: dev/com.ibm.ws.cdi.beansxml_fat/fat/src/com/ibm/ws/cdi12/fat/tests/BeanDiscoveryModeNoneTest.java [hunk 2]
```diff
@@ -26,28 +27,34 @@
...
+import com.ibm.ws.fat.util.browser.WebBrowserException;
+import com.ibm.ws.fat.util.browser.WebBrowserFactory;
+import com.ibm.ws.fat.util.browser.WebResponse;
 
 import componenttest.annotation.ExpectedFFDC;
 import componenttest.custom.junit.runner.Mode;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.impl.LibertyServerFactory;
 
 public class BeanDiscoveryModeNoneTest extends LoggingTest {
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.beansxml_fat/fat/src/com/ibm/ws/cdi12/fat/tests/BeanDiscoveryModeNoneTest.java:44]: I think you need `@RunWith(FATRunner.class)` here

File: dev/com.ibm.ws.cdi.beansxml_fat/fat/src/com/ibm/ws/cdi12/fat/tests/BeanDiscoveryModeNoneTest.java [hunk 3]
```diff
@@ -56,19 +63,25 @@ public static Archive buildShrinkWrap() {
      * @see com.ibm.ws.fat.LoggingTest#getSharedServer()
      */
     @Override
-    protected ShrinkWrapSharedServer getSharedServer() {
-        return SHARED_SERVER;
+    protected SharedServer getSharedServer() {
+        return null;
     }
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.beansxml_fat/fat/src/com/ibm/ws/cdi12/fat/tests/BeanDiscoveryModeNoneTest.java:68]: If you're not using `getSharedServer()` then I guess you're not using any of the methods on `LoggingTest` either?

In that case you can probably stop extending `LoggingTest`.

File: dev/com.ibm.ws.cdi.beansxml_fat/fat/src/com/ibm/ws/cdi12/fat/tests/BeanDiscoveryModeNoneTest.java [hunk 4]
```diff
@@ -56,19 +63,25 @@ public static Archive buildShrinkWrap() {
...
+
+        if (actualStatusCode != statusCode) {
+            throw new WebBrowserException("The expected status code was not thrown.  Expected " + statusCode + ", was " + actualStatusCode);
+        }
     }
 
     @AfterClass
     public static void tearDown() throws Exception {
-        if (SHARED_SERVER != null && SHARED_SERVER.getLibertyServer().isStarted()) {
+        if (server != null && server.isStarted()) {
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.beansxml_fat/fat/src/com/ibm/ws/cdi12/fat/tests/BeanDiscoveryModeNoneTest.java:84]: I don't think you need this `if` block (`server` will never be `null` and if the server is not started then `stopServer()` will just print a message and return)

File: dev/com.ibm.ws.cdi.beansxml_fat/fat/src/com/ibm/ws/cdi12/fat/tests/BeanDiscoveryModeNoneTest.java [hunk 5]
```diff
@@ -56,19 +63,25 @@ public static Archive buildShrinkWrap() {
...
     @ExpectedFFDC({ "javax.servlet.UnavailableException", "com.ibm.wsspi.injectionengine.InjectionException" })
     public void testcorrectExceptionThrown() throws Exception {
-        this.verifyStatusCode("/beanDiscoveryModeNone/TestServlet", 404);
+        WebBrowser webBrowser = WebBrowserFactory.getInstance().createWebBrowser((File) null);
+        int statusCode = 404;
+        int actualStatusCode = webBrowser.request(createURL("/beanDiscoveryModeNone/TestServlet"), statusCode);
+
+        if (actualStatusCode != statusCode) {
+            throw new WebBrowserException("The expected status code was not thrown.  Expected " + statusCode + ", was " + actualStatusCode);
+        }
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.beansxml_fat/fat/src/com/ibm/ws/cdi12/fat/tests/BeanDiscoveryModeNoneTest.java:79]: We have a class `HttpRequest` which can create the URL for you and is a little simpler.

I think you'd do
```
new HttpRequest(server, "/beanDiscoveryModeNone/TestServlet").expectCode(404).run(String.class);
```

**Comment by benjamin-confino** [comment]
#build

**Review by Azquelt — APPROVED** [review]

**Comment by benjamin-confino** [comment]
#run-libby-bot

**Pull request merged**